api/src/test/java/org/openmrs/api/AdministrationServiceTest.java

```diff
@@ -927,4 +928,59 @@ public void getPresentationLocales_shouldReturnLanguageLocaleIfItIsSpecifiedInAl
 		Assert.assertTrue("en", presentationLocales.contains(new Locale("en")));
 		Assert.assertTrue("es", presentationLocales.contains(new Locale("es")));
 	}
+	
+	/**
+	 * @see AdministrationService#getPresentationLocales()
+	 */
+	@Test
+	@Verifies(value = "should preserve insertion in Set returned by method", method = "getPresentationLocales()")
+	public void getPresentationLocales_shouldPreserveInsertionInSetReturnedByMethod()
+			throws Exception {
+		String globalPropertyLocaleListAllowedData = "en_GB, es, ja_JP, it_IT, pl_PL";
+		//The order of languages and locales is described above and should be followed bt `presentationLocales` Set
+		Context.getAdministrationService().saveGlobalProperty(
+				new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_LOCALE_ALLOWED_LIST, globalPropertyLocaleListAllowedData));
+		
+		List<Locale> locales = new ArrayList<Locale>();
+		//Add data in random order and verify that order is maintained in the end by checking against order in global property
+		locales.add(new Locale("pl", "PL"));
+		locales.add(new Locale("es"));
+		locales.add(new Locale("en"));
+		locales.add(new Locale("it", "IT"));
+		
+		MutableResourceBundleMessageSource mutableResourceBundleMessageSource = Mockito
+				.mock(MutableResourceBundleMessageSource.class);
+		Mockito.when(mutableResourceBundleMessageSource.getLocales()).thenReturn(locales);
+		
+		MutableMessageSource mutableMessageSource = Context.getMessageSourceService().getActiveMessageSource();
+		Context.getMessageSourceService().setActiveMessageSource(mutableResourceBundleMessageSource);
+		
+		Set<Locale> presentationLocales = Context.getAdministrationService().getPresentationLocales();
```

Comment:
From here onwards there is way too much unnecessary logic for the test. Just do something like:

List<Locale> presentationLocales = new ArrayList<Locale>(Context.getAdministrationService().getPresentationLocales());
assertThat(presentationLocales.get(0), is(...));
assertThat(presentationLocales.get(1), is(...));
// etc